### PR TITLE
coverity 1021735 : address constructor issue

### PR DESCRIPTION
**Pull request by dragon512**

### CODE REVIEW

**Comment by atsci** [comment]
Linux build *failed*! https://ci.trafficserver.apache.org/job/linux-github/2012/

**Comment by atsci** [comment]
clang format *failed*! https://ci.trafficserver.apache.org/job/clang-format-github/424/

**Comment by atsci** [comment]
RAT check *successful*! https://ci.trafficserver.apache.org/job/RAT-github/434/

**Comment by atsci** [comment]
AU check *successful*! https://ci.trafficserver.apache.org/job/autest-github/420/

**Comment by atsci** [comment]
FreeBSD11 build *successful*! https://ci.trafficserver.apache.org/job/freebsd-github/2117/

**Comment by atsci** [comment]
Intel CC build *successful*! https://ci.trafficserver.apache.org/job/icc-github/547/

**Comment by atsci** [comment]
Linux build *failed*! https://ci.trafficserver.apache.org/job/linux-github/2014/

**Comment by atsci** [comment]
clang format *successful*! https://ci.trafficserver.apache.org/job/clang-format-github/426/

**Comment by atsci** [comment]
RAT check *successful*! https://ci.trafficserver.apache.org/job/RAT-github/436/

**Comment by atsci** [comment]
clang-analyzer build *successful*! https://ci.trafficserver.apache.org/job/clang-analyzer-github/678/

**Comment by atsci** [comment]
AU check *successful*! https://ci.trafficserver.apache.org/job/autest-github/422/

**Comment by atsci** [comment]
FreeBSD11 build *successful*! https://ci.trafficserver.apache.org/job/freebsd-github/2119/

**Comment by atsci** [comment]
Intel CC build *successful*! https://ci.trafficserver.apache.org/job/icc-github/549/

**Comment by zwoop** [comment]
Build failure on Linux.

**Comment by atsci** [comment]
clang-analyzer build *successful*! https://ci.trafficserver.apache.org/job/clang-analyzer-github/680/

**Comment by zwoop** [comment]
[approve ci]

**Comment by atsci** [comment]
Linux build *failed*! https://ci.trafficserver.apache.org/job/linux-github/2049/

**Comment by atsci** [comment]
clang format *successful*! https://ci.trafficserver.apache.org/job/clang-format-github/460/

**Comment by atsci** [comment]
RAT check *successful*! https://ci.trafficserver.apache.org/job/RAT-github/470/

**Comment by atsci** [comment]
FreeBSD11 build *successful*! https://ci.trafficserver.apache.org/job/freebsd-github/2153/

**Comment by atsci** [comment]
AU check *successful*! https://ci.trafficserver.apache.org/job/autest-github/458/

**Comment by atsci** [comment]
Intel CC build *successful*! https://ci.trafficserver.apache.org/job/icc-github/583/

**Comment by atsci** [comment]
clang-analyzer build *successful*! https://ci.trafficserver.apache.org/job/clang-analyzer-github/714/

**Comment by atsci** [comment]
Linux build *failed*! https://ci.trafficserver.apache.org/job/linux-github/2063/

**Comment by atsci** [comment]
clang format *failed*! https://ci.trafficserver.apache.org/job/clang-format-github/475/

**Comment by atsci** [comment]
RAT check *successful*! https://ci.trafficserver.apache.org/job/RAT-github/484/

**Comment by atsci** [comment]
AU check *failed*! https://ci.trafficserver.apache.org/job/autest-github/472/

**Comment by atsci** [comment]
Intel CC build *successful*! https://ci.trafficserver.apache.org/job/icc-github/598/

**Comment by atsci** [comment]
FreeBSD11 build *successful*! https://ci.trafficserver.apache.org/job/freebsd-github/2167/

**Comment by atsci** [comment]
clang-analyzer build *successful*! https://ci.trafficserver.apache.org/job/clang-analyzer-github/727/

**Comment by atsci** [comment]
RAT check *successful*! https://ci.trafficserver.apache.org/job/RAT-github/496/

**Comment by atsci** [comment]
clang format *failed*! https://ci.trafficserver.apache.org/job/clang-format-github/487/

**Comment by atsci** [comment]
AU check *successful*! https://ci.trafficserver.apache.org/job/autest-github/484/

**Comment by atsci** [comment]
Linux build *successful*! https://ci.trafficserver.apache.org/job/linux-github/2075/

**Comment by atsci** [comment]
FreeBSD11 build *successful*! https://ci.trafficserver.apache.org/job/freebsd-github/2179/

**Comment by atsci** [comment]
Intel CC build *successful*! https://ci.trafficserver.apache.org/job/icc-github/610/

**Comment by atsci** [comment]
clang format *successful*! https://ci.trafficserver.apache.org/job/clang-format-github/490/

**Comment by atsci** [comment]
RAT check *successful*! https://ci.trafficserver.apache.org/job/RAT-github/499/

**Comment by atsci** [comment]
AU check *successful*! https://ci.trafficserver.apache.org/job/autest-github/487/

**Comment by atsci** [comment]
Intel CC build *successful*! https://ci.trafficserver.apache.org/job/icc-github/612/

**Comment by atsci** [comment]
clang-analyzer build *failed*! https://ci.trafficserver.apache.org/job/clang-analyzer-github/739/

**Comment by atsci** [comment]
Linux build *successful*! https://ci.trafficserver.apache.org/job/linux-github/2078/

**Comment by atsci** [comment]
Intel CC build *successful*! https://ci.trafficserver.apache.org/job/icc-github/614/

**Comment by atsci** [comment]
clang-analyzer build *successful*! https://ci.trafficserver.apache.org/job/clang-analyzer-github/740/

**Comment by atsci** [comment]
clang-analyzer build *successful*! https://ci.trafficserver.apache.org/job/clang-analyzer-github/743/

**Comment by atsci** [comment]
FreeBSD11 build *successful*! https://ci.trafficserver.apache.org/job/freebsd-github/2182/

File: iocore/net/UnixUDPNet.cc [hunk 1]
```diff
@@ -199,23 +201,21 @@ class UDPReadContinuation : public Continuation
...
-  int ifd;           // poll fd index
-  ink_hrtime period; // polling period
-  ink_hrtime elapsed_time;
-  ink_hrtime timeout_interval;
+  Ptr<IOBufferBlock> readbuf{nullptr};
+  int readlen                   = 0;
+  struct sockaddr_in6 *fromaddr = nullptr;
+  socklen_t *fromaddrlen        = nullptr;
+  int fd                        = ~0; // fd we are reading from
+  int ifd                       = ~0; // poll fd index
```
Comment (zwoop) [inline iocore/net/UnixUDPNet.cc]: Is this intentional? ~0 ?

Comment (SolidWallOfCode) [inline iocore/net/UnixUDPNet.cc]: Should probably be `ts::NO_FD`.

**Comment by atsci** [comment]
RAT check *successful*! https://ci.trafficserver.apache.org/job/RAT-github/546/

**Pull request merged**